app/utils/kci_test/regressions.py

```diff
@@ -0,0 +1,466 @@
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Logic to find regressions in test groups reports."""
+
+import bson
+import redis
+
+import models
+import utils
+import utils.database.redisdb as redisdb
+import utils.db
+
+# How the key to access the regressions data structure is formatted.
+# Done in this way so that we can use mongodb dot notation to access embedded
+# documents.
+REGRESSION_FMT = "{:s}.{:s}.{:s}.{:s}.{:s}.{:s}"
```

Comment:
Having named elements would be a bit more robust and self-documenting: `"{lab:s}.{arch:s}.`.

Reply:
I haven't checked but  that'll be problematic if the different functions using REGRESSION_FMT use different parameters. This comes from `app/utils/boot/regressions.py` and eventually we would want to use the same format for both kind of regressions.